pages/api/teammatching/subscription/delete.ts

```diff
@@ -0,0 +1,68 @@
+import { firestore } from 'firebase-admin';
+import { NextApiRequest, NextApiResponse } from 'next';
+import initializeApi from '../../../../lib/admin/init';
+import { userIsAuthorized } from '../../../../lib/authorization/check-authorization';
+
+initializeApi();
+const db = firestore();
+
+// interface of subscription data that contains the user id and the post id
+interface SubscriptionData {
+  userId: string;
+  postId: string;
+}
+
+// delete one subscription for user that unsubscribe them to a post
+async function deleteSubscription(req: NextApiRequest, res: NextApiResponse) {
```

Comment:
I think we'll need a way to verify whether request initiator matches user id provided in the request body.